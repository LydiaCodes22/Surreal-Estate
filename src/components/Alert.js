import React from "react";
import "../styles/alert.css";
import PropTypes from "prop-types";

const Alert = ({ statusCode }) => {
  if (statusCode === 201) {
    return (
      <div className="success-alert">
        Success, your property has been created
      </div>
    );
  }
  if (statusCode === 404) {
    return (
      <div className="server-error-alert">
        {" "}
        Server error. Please try again later.
      </div>
    );
  }
  return <div> </div>;
};

Alert.propTypes = {
  statusCode: PropTypes.number.isRequired,
};

export default Alert;
