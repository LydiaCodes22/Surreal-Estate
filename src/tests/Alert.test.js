import { render, screen } from "@testing-library/react";
import React from "react";
import Alert from "../components/Alert";

describe("Alert", () => {
  const validStatus = 201;
  const badStatus = 404;
  it("matches the snapshot", () => {
    const { asFragment } = render(<Alert statusCode="" />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("renders a success message when the user creates a property", () => {
    render(<Alert statusCode={validStatus} />);
    expect(screen.getByText(/success/i)).toBeInTheDocument();
  });
  it("renders an error message when the user creates a property", () => {
    render(<Alert statusCode={badStatus} />);
    expect(screen.getByText(/error/i)).toBeInTheDocument();
  });
});
