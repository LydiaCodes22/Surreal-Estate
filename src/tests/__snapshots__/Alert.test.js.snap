// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="error-alert"
  >
     Not successful
  </div>
</DocumentFragment>
`;
